File: ChildrenInSubtree/children_in_subtree.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node Node;
struct Node {
    int data;
    Node *left;
    Node *right;
};

int children(Node* head) {
    if(!head) return 0;
    head->data = 1 + children(head->left) + children(head->right);
    return head->data;
}


// Tester
int main() {
    Node A, B, C, D, E;
    A.left = &B;
    A.right = &C;
    B.left = &D;
    B.right = 0;
    C.left = 0;
    C.right = 0;
    D.left = 0;
    D.right = &E;
    E.left = 0;
    E.right = 0;

    children(&A);
    printf("A has %d children\n", A.data);
}

File: ChildrenInSubtree/children_in_subtree.py
class Node:
    left = 0
    right = 0

def childCount(head):
    if not head:
        return 0
    return 1 + childCount(head.left) + childCount(head.right)

A = Node()
B = Node()
C = Node()
D = Node()
E = Node()
F = Node()
G = Node()

A.left = B
A.right = C
B.left = D
B.right = G
C.right = F
D.right = E

"""
          A
       B     C
     D   G     F
      E
"""

print "Child count of A is: " + str(childCount(A))
print "Child count of B is: " + str(childCount(B))
print "Child count of C is: " + str(childCount(C))
print "Child count of D is: " + str(childCount(D))
print "Child count of E is: " + str(childCount(E))
print "Child count of F is: " + str(childCount(F))
print "Child count of G is: " + str(childCount(G))
